### ISSUE TITLE
Postgres string arrays escaped characters

### ISSUE THREAD

**Issue by chamini2** [tgriesser/knex#1793]
I ran into what I think is a bug today. When saving a string array with tgriesser/bookshelf, if any of the strings contained newlines (`\n`), they would come back as `n`. For example, saving `['hello\n\nbye', 'hi\n']` would be fetched saved in the DB as `['hellonnbye', 'hin']`.

I tested some stuff in Postgres directly to see where was the problem, and it seems to me this [line](https://github.com/tgriesser/knex/blob/5890e4d4ea4304e1c651d5aa6690857e439f238e/src/dialects/postgres/index.js#L238) might be the problem. It checks if the element of the array is a string and doesn't escape it when it is, I believe there the function [`escapeString`](https://github.com/tgriesser/knex/blob/5890e4d4ea4304e1c651d5aa6690857e439f238e/src/dialects/postgres/index.js#L57) should be called, right?

Since I'm not sure, I'll leave it up to you, but wanted to report the issue! 😄

**Comment by elhigu** [comment]
Hmm.. looks correct to me

```js
> knex = require('knex')({ client: 'pg' });
> knex('table').insert({ strarray: ['foo\nbar'], str: 'foo\nbar' }).toSQL()
{ method: 'insert',
  options: {},
  timeout: false,
  cancelOnTimeout: false,
  bindings: [ 'foo\nbar', [ 'foo\nbar' ] ],
  __knexQueryUid: '0297c0a5-124f-4efc-a6a9-1fa22da7c99c',
  sql: 'insert into "table" ("str", "strarray") values (?, ?)',
  returning: undefined }
> 
```

Or is does this consider some other issue? Could you add some example to see the error you encoutered?

**Comment by chamini2** [comment]
Ok, I did not expect that to work. Maybe it has to do with bookshelf?

I did the following in bookshelf:

```js
this.on('saving', function(model) {
  console.log(model.get('array_col');
});

this.on('fetched', function(model) {
  console.log(model.get('array_col');
});
```

and got:

```
[ 'Some interesting stuff.\nFollowed by an explanation.\n', 'Another item.\n' ]
[ 'Some interesting stuff.nFollowed by an explanation.n', 'Another item.n' ]
```

**Comment by elhigu** [comment]
yep, better to try to find answer there for now. I have no idea how bookshelf creates its queries.

**Comment by chamini2** [comment]
Maybe this was related to #1710 ?

**Comment by elhigu** [comment]
@chamini2 Doesn't look like the same issue. Which database are you using? I just noticed that I tried to reproduce it only with postgresql.

**Comment by chamini2** [comment]
The thing is, bookshelf creates its queries with Knex. What I think is that maybe when #1710 was fixed in #1731, this was fixed, and I was using an old version. I will confirm if this is fixed in the latest knex and reopen if not. Thank you!

PD: I was always using Postgres.
